lib/ConstantFraction.js

```diff
@@ -176,13 +212,34 @@ ConstantFraction.addConstantAndFraction = function(node) {
     newNode.args[0] = newFractionNode;
     newNode.args[1] = newConstNode;
   }
-  return NodeStatus.nodeChanged(changeType, oldNode, newNode);
+
+  let subSteps = [NodeStatus.nodeChanged(changeType, oldNode, newNode)];
+  // The newNode for the last step is now oldNode for next step.
+  oldNode = NodeStatus.resetChangeGroups(newNode);
+
+  // If we changed an integer to a fraction, we need to add the steps for
+  // adding the fractions.
+  if (changeType === MathChangeTypes.CONVERT_INTEGER_TO_FRACTION) {
+    const addFractionStatus = ConstantFraction.addConstantFractions(oldNode);
+    subSteps = subSteps.concat(addFractionStatus.subSteps);
+  }
+  // Otherwise, add the two constants
+  else {
+    newNode = NodeCreator.constant(oldNode.eval());
+    subSteps.push(NodeStatus.nodeChanged(
+      MathChangeTypes.SIMPLIFY_ARITHMETIC, oldNode, newNode));
+  }
+
+  const lastStep = subSteps[subSteps.length - 1];
+  newNode = NodeStatus.resetChangeGroups(lastStep.newNode);
+
+  return NodeStatus.nodeChanged(changeType, node, newNode, true, subSteps);
 };
 
 // Given a list of nodes `fractionNodes` that all have the same denominator,
 // add them together. e.g. 2/3 + 5/3 -> (2+5)/3
 // Returns the new node.
-function addFractionsWithSameDenominator(fractionNodes) {
+function writeNumeratorsAboveCommonDenominator(fractionNodes) {
```

Comment:
I think I got the message from some other app

combine makes more sense, especially at this level